server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java

```diff
@@ -197,6 +202,55 @@ void browseShouldReturnCurrentlyEnqueuedMailFromAllSlices() throws Exception {
                 "5-1", "5-2", "5-3", "5-4", "5-5");
         }
 
+        @RepeatedTest(50)
+        void browseStartShouldBeUpdated(CassandraCluster cassandraCluster) {
+            int emailCount = 250;
+
+            StatementRecorder.Selector selector = preparedStatementStartingWith("UPDATE browsestart");
+            StatementRecorder statementRecorder = cassandraCluster.getConf()
+                .recordStatements(selector);
+
+            clock.setInstant(IN_SLICE_1);
+            enqueueSomeMails(namePatternForSlice(1), emailCount);
+            dequeueMails(emailCount);
+
+            clock.setInstant(IN_SLICE_2);
+            enqueueSomeMails(namePatternForSlice(2), emailCount);
+            dequeueMails(emailCount);
+
+            clock.setInstant(IN_SLICE_3);
+            enqueueSomeMails(namePatternForSlice(3), emailCount);
+            dequeueMails(emailCount);
+
+            // The actual rate of update should actually be lower than the update probability.
+            assertThat(statementRecorder.listExecutedStatements(selector))
+                .hasSizeBetween(2, 12);
+        }
```

Comment:
Basically, this test asserts a certain statement on something that depends on probability. It means it always has a chance to fail.
We even can calculate (theoretically) the failure percentage of this test using the Bernoulli trial, and the result is not small.
I suggest we mark these tests as `@Unstable`. WDYT @Arsnael @chibenwa ?

Reply:
I'm not sure. I understand your comment and it makes sense. But this test seemed pretty stable to me before this change, now it's very unstable. 

Also it feels different things are being performed in this PR, could potentially cut it smaller?

Reply:
> But this test seemed pretty stable to me before this change, now it's very unstable.

No.
https://github.com/apache/james-project/pull/1623#issuecomment-1639303055

Reply:
> We even can calculate (theoretically) the failure percentage of this test using the Bernoulli trial, and the result is not small.

That's the point: the test is here to ensure we avoid updating the browse start when not needed.

Upper value verifies that if current browseStart is still in the current slice we do not update. BrowseStart updates count should thus be '3' or '4' but exactly the count of slice switches.

Now the rabbitMQ prefetch fucks this up by introducing concurrency issues. Setting the QOS to 1 solved the issue. We can't set this up for the all test suite but using a QOS of 1 in this test would INDEED solve our issue.

Cc @quantranhong1999

Reply:
Thank you for your suggestion. It indeed helps the tests to be more stable.